### PR TITLE
Update type mapping for Iceberg connector

### PR DESCRIPTION
**Pull request by Jessie212**
## Description

> Is this change a fix, improvement, new feature, refactoring, or other?

Improvement.

> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)

Changes made to documentation.

> How would you describe this change to a non-technical end user or system administrator?

Add fragments for Avro, ORC, and Parquet mappings to type mapping section.

## Documentation

( ) No documentation is needed.
(x) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

(x) No release notes entries required.
( ) Release notes entries required with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: docs/src/main/sphinx/connector/avro-type-mapping.fragment [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  :header-rows: 1
+
+  * - Trino type
+    - Avro type
+    - Notes
+  * - ``BOOLEAN``
+    - ``BOOLEAN``
+    -
+
+No to other types are supported.
```
Comment (jhlodin) [inline docs/src/main/sphinx/connector/avro-type-mapping.fragment:37]: No other

File: docs/src/main/sphinx/connector/avro-type-mapping.fragment [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  * - ``BOOLEAN``
+    - ``BOOLEAN``
+    -
+
+No other types are supported
+
+Trino to Avro type mapping
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The connector maps Trino| types to the corresponding Avro types following
```
Comment (jhlodin) [inline docs/src/main/sphinx/connector/avro-type-mapping.fragment:23]: Remove |

**Review by jhlodin — CHANGES_REQUESTED** [review]
With this approach we're looking at up to 6 extremely vertical tables for three already long connector pages, are we sure there's not a more compact approach? Like the usual two tables for each direction, but with additional column(s) describing support for each file format?

**Comment by jhlodin** [comment]
> > With this approach we're looking at up to 6 extremely vertical tables for three already long connector pages, are we sure there's not a more compact approach? Like the usual two tables for each direction, but with additional column(s) describing support for each file format?
> 
> The tables, as they are, are already quite squished. Adding additional rows might make the text unreadable, especially if there's a notes column.

@Jessie212 Horizontally squished? We could just adjust the column widths if we need to give the text breathing room

**Comment by alexjo2144** [comment]
Can we just add links to
- Avro: https://iceberg.apache.org/spec/#avro
- Parquet: https://iceberg.apache.org/spec/#parquet
- ORC: https://iceberg.apache.org/spec/#orc 

And leave the existing table?

**Comment by Jessie212** [comment]
> Can we just add links to
> 
>     * Avro: https://iceberg.apache.org/spec/#avro
> 
>     * Parquet: https://iceberg.apache.org/spec/#parquet
> 
>     * ORC: https://iceberg.apache.org/spec/#orc
> 
> 
> And leave the existing table?

Are types in the **Type** column Trino data types?

**Comment by Jessie212** [comment]
To make sure I understand: 

1) The unspecified "Type" in the Apache link is actually Iceberg: Iceberg -> Avro, Iceberg -> Parquet, Iceberg -> ORC?

2) In our docs, we substitute the Trino types with Iceberg so the table shows Iceberg to file type only? No mention of Trino?

**Comment by Jessie212** [comment]
To make sure I understand: 

1) The unspecified "Type" in the Apache link is actually Iceberg: Iceberg -> Avro, Iceberg -> Parquet, Iceberg -> ORC?

2) In our docs, we substitute the Trino types with Iceberg so the table shows Iceberg to file type only? No mention of Trino?

**Comment by alexjo2144** [comment]
> The unspecified "Type" in the Apache link is actually Iceberg: Iceberg -> Avro, Iceberg -> Parquet, Iceberg -> ORC?

Yes

> In our docs, we substitute the Trino types with Iceberg so the table shows Iceberg to file type only? No mention of Trino?

The current documentation shows Trino <-> Iceberg

**Comment by Jessie212** [comment]
> The current documentation shows Trino <-> Iceberg

What I meant was, we should modify the current docs to show Iceberg in the place of Trino.

File: docs/src/main/sphinx/connector/iceberg.rst [hunk 3]
```diff
@@ -374,75 +386,69 @@ Iceberg to Trino type mapping:
...
+  * - ``FIXED (L)``
+    - ``BINARY``
+  * - ``BINARY``
+    - ``BINARY``
   * - ``STRUCT(...)``
     - ``ROW(...)``
   * - ``LIST(e)``
     - ``ARRAY(e)``
-  * - ``MAP(k,v)``
+  * - ``MAP``
```
Comment (alexjo2144) [inline docs/src/main/sphinx/connector/iceberg.rst:397]: The k, v here was important, I'd leave that in
```suggestion
  * - ``MAP(k,v)``
```

File: docs/src/main/sphinx/connector/iceberg.rst [hunk 4]
```diff
@@ -374,75 +386,69 @@ Iceberg to Trino type mapping:
...
   * - ``UUID``
     - ``UUID``
-    -
   * - ``VARBINARY``
     - ``BINARY``
-    -
+  * - ``BINARY``
+    - ``FIXED (L)``
+  * - ``BINARY``
+    - ``BINARY``
```
Comment (alexjo2144) [inline docs/src/main/sphinx/connector/iceberg.rst:443]: Trino doesn't have a Binary type. No Trino types map to the Iceberg `Fixed` type.
```suggestion
```

File: docs/src/main/sphinx/connector/iceberg.rst [hunk 5]
```diff
@@ -374,75 +386,69 @@ Iceberg to Trino type mapping:
...
   * - ``ROW(...)``
     - ``STRUCT(...)``
-    - All fields must have a name
   * - ``ARRAY(e)``
     - ``LIST(e)``
-    -
   * - ``MAP(k,v)``
-    - ``MAP(k,v)``
-    -
+    - ``MAP``
```
Comment (alexjo2144) [inline docs/src/main/sphinx/connector/iceberg.rst:449]: Here also, I think the k, v is important
```suggestion
    - ``MAP(k, v)``
```

File: docs/src/main/sphinx/connector/iceberg.rst [hunk 6]
```diff
@@ -374,75 +386,69 @@ Iceberg to Trino type mapping:
     - ``UUID``
   * - ``BINARY``
     - ``VARBINARY``
+  * - ``FIXED (L)``
+    - ``BINARY``
```
Comment (alexjo2144) [inline docs/src/main/sphinx/connector/iceberg.rst:390]:
```suggestion
    - ``VARBINARY``
```

File: docs/src/main/sphinx/connector/iceberg.rst [hunk 7]
```diff
@@ -374,75 +386,69 @@ Iceberg to Trino type mapping:
...
+  * - ``BINARY``
+    - ``BINARY``
```
Comment (alexjo2144) [inline docs/src/main/sphinx/connector/iceberg.rst:392]: This is a duplicate of the Binary -> Varbinary two lines above
```suggestion
```

**Review by alexjo2144 — APPROVED** [review]
Reads well to me

**Comment by findepi** [comment]
Thanks @alexjo2144 for your review

**Pull request merged**